### DISCUSSION TITLE
Add required PHP modules to composer.json

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#2174] [Ideas]
as we absolutely require some php extensions like mbstring, we should make sure this is visible in the composer.json

**Comment by luigifab** [comment]
Here is what PHPStorm propose for app directory:
```
+   "ext-curl": "*",   (php-curl)
+   "ext-gd": "*",   (php-gd)
+   "ext-json": "*",   (php-json)
+   "ext-mbstring": "*",   (php-mbstring)
+   "ext-pdo": "*",   (php-mysql)
+   "ext-simplexml": "*",   (php-xml)
+   "ext-dom": "*",   (php-xml)
    "ext-libxml": "*",
+   "ext-soap": "*",   (php-soap)
    "ext-iconv": "*",
    "ext-zlib": "*",
    "ext-ftp": "*",
```

**Comment by Flyingmana** [comment]
now we need to find out, which of them are optional :/
Iam pretty sure the ftp one is not generally needed?

**Comment by elidrissidev** [comment]
Mage_Dataflow is using `ext-ftp`, and I think Mage_Backup too but it's disabled by default.

**Comment by luigifab** [comment]
I updated my anwser by adding php package name.
Not sure who provides "ext-libxml", perhaps php-xml?
I checked my packages, php-iconv is not installed.
I don't remember if php-soap is required or not.

**Comment by elidrissidev** [comment]
> I don't remember if php-soap is required or not.

Yes it is, it's used by Mage_Api and I some other core modules.

**Comment by fballiano** [comment]
I've created a PR for this

**Comment by Flyingmana** [comment]
@ADDISON74 how is this a discussion? Its a very well defined and scoped Task which needs/should be done

**Reply by addison74** [reply]
@Flyingmana - Until the Discussions section existed the Issues section included absolutely everything. From requests for help, ideas, issues and much more. So that we have reached an impressive number of open posts almost impossible to manage until recently. Today in the Issues section there are only reported issues related to functionality. Some require testing/investigation, others are confirmed, others need PRs.

When the Discussions section was enabled I was the only one who moved some of the open posts from Issues to various sections of Discussions. Then I followed a rule, when I found obvious issues that required immediate information and quick resolution I reported them in Issues and accompanied them with detailed descriptions. When I promoted ideas to change or improve something I either created PRs or posted them in Discussions > General/Ideas when I thought it was necessary to get more opinions until we decide the best solution. 

Now let me answer your question. Your post does not fall under Issues even if it is well defined. Indeed it is about a request that must be made but that requires discussions after which there should be a solution. What makes the difference between an real issue and an idea? My latest OM will not run properly due to the issue, instead it has no serious issue in the absence of the idea.
